src/lib.rs

```diff
@@ -783,6 +801,20 @@ fn user_create(
 fn initialize(init: &DeviceContext) -> Result<IronOxide, String> {
     Ok(ironoxide::initialize(init)?)
 }
+fn initialize_and_rotate(init: &DeviceContext, password: &str) -> Result<IronOxide, String> {
```

Comment:
Craig and I discussed this offline, but it's probably worth noting here for other reviewers and posterity...

This is somewhat different from ironoxide's interface in that it does not allow a caller to opt for rotation checks and then either (a) not do them, or (b) only do select rotations. Due to the fact that the `IronOxide` struct is not cloneable and there is no way in rust_swig consume `self` there does not seem to be a straightforward way to extract the `IronOxide` instance out of a rust-generated struct.

If rust_swig gets the ability to take `self` instead of `&self`, I think we could "exchange" a struct holding an `IronOxide` for an `IronOxide`. If we want to allow customized behavior we could also expand the signature of this initialize function with handlers of some sort for doing the user/group private key rotations.

Or maybe there's a solution I'm overlooking. /shrug